### DISCUSSION TITLE
Dicas para criar a versão de um site em outros idiomas

### DISCUSSION THREAD

**Discussion by ghost** [frontendbr/forum#881] [Perguntas e Respostas]
Olá pessoal, 

Hoje me deparei novamente com a tarefa de criar uma versão de um site em outro idioma.

Nesses casos, costumo criar duas versões do mesmo código, cada uma em um idioma. Porém é uma prática que não me parece muito elegante e que gera problemas de manutenção já que as páginas ficam duplicadas.

Eu estava imaginando se não existiria uma solução melhor para esta questão.

Alguém tem alguma sugestão?

**Comment by mahenrique94** [comment]
Procure sobre internacionalização @andressabertolini, onde você armazena todas as escritas do site em um arquivo, ai tem um arquivo para cada idioma e na página você chama as escritas pela chave do arquivo.

**Comment by simaodeveloper** [comment]
@mahenrique94 mas se ela estiver pensando no SEO, seria legal automatizar essa parte para gerar as páginas estáticamente, assim ela não perde ranking no google da vida.

Caso for um APP então poderia ser algo dinâmico, utilizando por exemplo https://braziljs.org/blog/implementando-internacionalizacao-i18n-com-javascript/

**Comment by mahenrique94** [comment]
nunca tive problemas com SEO utilizando internacionalização @simaodeveloper, depois irei ler sobre o assunto.

**Comment by simaodeveloper** [comment]
@mahenrique94 você implementa via back-end? se sim, então é outra história. Achei esses slides aqui, 
acho que vão ajudar http://yycjs.com/seo-i18n/

**Comment by mahenrique94** [comment]
sim @simaodeveloper, achei estranho, mas foi apenas um mal entendido hehe, no front até o momento nunca precisei, vou dar uma olhada nos slides, valeu

**Comment by shadowmaru** [comment]
Dá pra implementar no backend ou no frontend, aí depende de como o site está estruturado e como  HTML é gerado. No Rails, por exemplo, já está tudo preparado pra isso. No site da Smart Fit, o projeto é o mesmo com os locales pra México, Chile, República Dominicana, Colômbia, Peru e Equador. No React, tem o 'react-i18next'. Tem também um jeito do JavaScript usar as traduções do Rails, aí é um híbrido.

**Comment by Sergioamjr** [comment]
Implementamos essa lib em React aqui no trabalho https://github.com/yahoo/react-intl, funcionou bem, mas não sei como ficou o SEO porque a aplicação não é aberta pra todo mundo.

**Comment by dann95** [comment]
@andressabertolini a sugestão deve vir em base ao que você usa, o que você usa de stack nesse site, ele é estático? é renderizado serverside? é um spa? é dai que surgem as soluções para traduzir.
